### ISSUE TITLE
Allow translating C code that conflicts with primitive types

### ISSUE THREAD

**Issue by daurnimator** [ziglang/zig#3756]
Excerpt from `zig translate-c` output when processing the linux kernel headers:
```zig
pub const __s8 = i8;
pub const __u8 = u8;
pub const __s16 = c_short;
pub const __u16 = c_ushort;
pub const __s32 = c_int;
pub const __u32 = c_uint;
pub const __s64 = c_longlong;
pub const __u64 = c_ulonglong;
pub const s8 = __s8;
pub const u8 = __u8;
pub const s16 = __s16;
pub const u16 = __u16;
pub const s32 = __s32;
pub const u32 = __u32;
pub const s64 = __s64;
pub const u64 = __u64;
```

Which ends up failing when attempting to use the resulting zig file:
```
cimport.zig:30:5: error: declaration shadows primitive type 'u8'
pub const u8 = __u8;
    ^
```

I'm not sure if the fix is something in translate-c itself, or if we need some new special zig syntax to allow it.

**Comment by Vexu** [comment]
Related #1802

**Comment by daurnimator** [comment]
Fixed via #3935 ... sort of. 
```C
typedef unsigned char u8;
```
Is now translated to 
```zig
pub const _u8 = u8;
```

However duplicates are not handled, e.g.
```C
typedef unsigned char u8;
typedef unsigned char _u8;
```
Is translated to 
```zig
pub const _u8 = u8;
pub const _u8 = u8;
```

which fails with 
```
error: redefinition of '_u8'
pub const _u8 = u8;
    ^
```
